modules/statistics/php/stats_mri.class.inc

```diff
@@ -106,34 +109,95 @@ class Stats_MRI extends \NDB_Form
             $subcat  = $row['Subcat'];
             $center  = $row['CenterID'];
 
+            /* Calculate the totals for each element of the stats table. Each
+             * node of this nested array contains a key-value array consisting
+             * of the subcategories as they keys and the totals as values.
+             * There is one additional key, 'total'.
+             * E.g.
+             *      array(
+             *          'total' = x;
+             *          $subcategory1 = y;
+             *          $subcategory2 = z;
+             *          ...
+             *      )
+             */
             if (in_array($vl, $visits)
                 && in_array($subcat, $subcats)
                 && $this->_inCenter($center, $centres)
             ) {
                 $C = 'C' . $center;
+                /* This array contains all the $subcategories as keys as well
+                * as 'total'. All values are 0.
+                */
+                $blankStatsArray = $this->initializeStatsArray($subcats);
+                /* Initialize values for all of the endpoints using the array
+                 * just created. This step is done to prevent undefined index
+                 * notices from appearing in the PHP logs for uninitialzed keys.
+                 */
+
+                // Data
+                if (!isset($tpl_data['data'])) {
```

Comment:
@johnsaigle I believe that @intralizee had submitted a PR with these verification in the code where needed.
I do think that it is cleaner to create all arrays in one place.

Reply:
I believe the PR you mentioned is the one I linked in the description, https://github.com/aces/Loris/pull/3928. This PR is replacing that one and her commits are included here.